pystac/stac_object.py

```diff
@@ -463,13 +463,17 @@ def from_file(
             The specific STACObject implementation class that is represented
             by the JSON read from the file located at HREF.
         """
+        if cls == STACObject:
+            return pystac.read_file(href)
+
         if stac_io is None:
             stac_io = pystac.StacIO.default()
 
         if not is_absolute_href(href):
             href = make_absolute_href(href)
 
-        o = stac_io.read_stac_object(href)
+        d = stac_io.read_json(href)
+        o = cls.from_dict(d, href=href, migrate=True)
```

Comment:
Why this change? Couldn't this remain `stac_io.read_stac_object` for one codepath to handle href -> STACObject logic?

The type will be `STACObject` as far as type hinting goes, but the underlying class will be either a Catalog, Collection or Item. The casting done in the implementations should work out fine.

Reply:
Just made this change and saw that some tests break with custom STACObjects. disregard.